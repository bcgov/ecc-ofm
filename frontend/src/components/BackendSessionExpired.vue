<template>
  <div style="display: none">
    <a id="logout_href" :href="link" />
  </div>
</template>

<script>
import { Routes } from '@/utils/constants'
import { useAuthStore } from '@/stores/auth'
import { mapActions } from 'pinia'
export default {
  name: 'BackendSessionExpired',
  data() {
    return {
      link: Routes.SESSION_EXPIRED
    }
  },
  mounted() {
    window.location = document.getElementById('logout_href').href
  },
  methods: {
    ...mapActions(useAuthStore, ['setJwtToken']),
    logout() {
      this.setJwtToken()
    }
  }
}
</script>
