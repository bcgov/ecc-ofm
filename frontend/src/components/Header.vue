<!-- eslint-disable vue/multi-word-component-names -->
<template>
  <v-app-bar absolute color="rgb(0, 51, 102)" class="sysBar pl-10 pr-10" style="z-index: 1002">
    <!-- Navbar content -->
    <a tabindex="-1" href="/">
      <img tabindex="-1" src="@/assets/images/bc-gov-logo.svg" class="logo" alt="B.C. Government Logo" />
    </a>
    <a tabindex="-1" href="/">
      <img class="verticalLine" />
    </a>
    <v-toolbar-title style="margin-left: 0px;">
      <h3>{{ appTitle }}</h3>
    </v-toolbar-title>
    <v-spacer></v-spacer>
    <div v-if="isAuthenticated && user">
      <v-menu name="user_options" offset-y>
        <template #activator="{ props }">
          <v-chip v-bind="props" tabindex="0" pill color="#003366" dark>
            <v-avatar left color="info">
              {{ user.userName[0] }}
            </v-avatar>
            <span class="display-name pl-1">{{ user.userName }}</span>
          </v-chip>
        </template>
        <v-list dark style="background-color: #003366; color: white">
          <v-list-item :href="routes.LOGOUT" id="logout_button" style="min-height: 4vh" title="Logout" />
        </v-list>
      </v-menu>
    </div>
  </v-app-bar>
</template>

<script>
import { mapActions, mapState } from 'pinia'
import { Routes } from '@/utils/constants'
import { useAuthStore } from '@/stores/auth'

export default {
  data() {
    return {
      appTitle: import.meta.env.VITE_APP_TITLE,
      routes: Routes,
      user: null
    }
  },
  created() {
      useAuthStore().getUserInfo().then(() => {
        this.user = this.userInfo
      })
  },
  computed: {
    ...mapState(useAuthStore, ['userInfo', 'isAuthenticated'])
  },
  methods: {
    ...mapActions(useAuthStore, ['getUserInfo']),
    useAuthStore
  }
}
</script>

<style>
.gov-header .v-icon {
  padding-left: 10px;
}

a {
  text-decoration: none;
}

.logo {
  padding-right: 15px;
  padding-top: 4px;
  width: 205px;
  height: 77px;
}

.verticalLine {
  border-left: 1px solid #DFB433;
  height: 50px;
  margin-left: 12px;
  padding-left: 24px;
  margin-top: 4px;
}

.gov-header .title {
  color: #fff;
  text-decoration: none;
}

.sysBar {
  border-bottom: 2px solid rgb(252, 186, 25) !important;
  z-index: 8;
}

.gov-header .v-btn,
.v-btn--active.title:before,
.v-btn.title:focus:before,
.v-btn.title:hover:before {
  color: #fff;
  background: none;
}

.secondary_color {
  background-color: var(--v-secondary-base);
}

.v-input__slot {
  padding-top: 10px;
}

.display-name {
  color: white;
}

.top-down {
  padding-top: 20px;
  height: 80%;
}

.mainTitle {
  font-size: 1.0rem;
}
</style>
