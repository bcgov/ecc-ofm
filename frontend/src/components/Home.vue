<!-- eslint-disable vue/multi-word-component-names -->
<template>
  <v-card
    title="OFM"
    subtitle="Temp Home Screen"
    text="..."
    style="height:100%;"
  ></v-card>
</template>
