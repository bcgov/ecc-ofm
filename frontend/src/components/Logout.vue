<template>
  <v-container fluid class="full-height">
    <!-- login article -->
    <article name="logout-banner">
      <v-row align="center" justify="center">
        <v-col xs="10" sm="10" md="8" lg="4" xl="3">
          <v-card class="session-expired-card">
            <v-card-title class="gov-header">
              <h4 id="logout_text">Logged Out</h4>
            </v-card-title>
            <v-card-text id="logout_descriptor">
              <v-row style="margin: 0.3rem"> You have Logged out. </v-row>
              <a
                id="login-button"
                :href="routes.LOGIN"
                class="ma-1"
                dark
                color="#003366"
                @click="clearStorage"
                >Log In</a
              ><span>again if you wish to continue.</span>
            </v-card-text>
          </v-card>
        </v-col>
      </v-row>
    </article>
  </v-container>
</template>

<script>
import { mapActions } from 'pinia'
import { Routes } from '../utils/constants'
import { useAuthStore } from '@/stores/auth'

export default {
  // eslint-disable-next-line vue/multi-word-component-names
  name: 'Logout',

  data() {
    return {
      routes: Routes
    }
  },
  created() {
    this.setJwtToken()
  },
  methods: {
    ...mapActions(useAuthStore, ['setJwtToken']),
    clearStorage() {
      this.setJwtToken()
    }
  }
}
</script>

<style scoped>
.full-height {
  height: 100%;
}
.session-expired-card {
  margin-top: 15rem;
  width: 100%;
  background: #d9e7d8;
}
</style>
