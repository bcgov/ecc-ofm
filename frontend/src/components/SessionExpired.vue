<template>
  <v-container fluid class="full-height">
    <!-- login article -->
    <article name="session-expired-banner" class="top-banner">
      <v-row align="center" justify="center">
        <v-col xs="10" sm="10" md="8" lg="4" xl="3">
          <v-card class="session-expired-card">
            <v-card-title class="gov-header">
              <h4 id="session-expired-text">Session Expired</h4>
            </v-card-title>
            <v-card-text id="session-expired-descriptor">
              <v-row style="margin: 0.3rem">
                Your secure session has ended as a result of inactivity.
              </v-row>
              <a
                id="login-button"
                :href="routes.LOGIN"
                class="ma-1"
                dark
                color="#003366"
                @click="clearStorage"
                >Log In</a
              ><span>again to continue.</span>
            </v-card-text>
          </v-card>
        </v-col>
      </v-row>
    </article>
  </v-container>
</template>

<script>
import { mapActions } from 'pinia'
import { Routes } from '../utils/constants'
import { useAuthStore } from '@/stores/auth'

export default {
  name: 'SessionExpired',

  data() {
    return {
      routes: Routes
    }
  },
  mounted() {
    this.setJwtToken()
  },
  methods: {
    ...mapActions(useAuthStore, ['setJwtToken']),
    clearStorage() {
      this.setJwtToken()
    }
  }
}
</script>

<style scoped>
.full-height {
  height: 100%;
}
.session-expired-card {
  margin-top: 15rem;
  width: 100%;
  background: #f2e8d5;
}
</style>
