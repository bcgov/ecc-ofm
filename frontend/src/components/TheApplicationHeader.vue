<template>
  <v-container v-if="isAuthenticated" fluid>
    <v-row>
      <v-col class="header-application justify-md-end justify-sm-start">
        {{ referenceNumber }}
      </v-col>
    </v-row>
  </v-container>
</template>
<script>
import { mapState } from 'pinia'
import { useAuthStore } from '@/stores/auth'

export default {
  props: {
    referenceNumber: {
      type: String,
      default: '',
    },
  },
  computed: {
    ...mapState(useAuthStore, ['isAuthenticated']),
  },
}
</script>
<style scoped>
.header-application {
  display: flex;
  color: #003366;
  font-weight: 600;
  font-size: 1.3em;
}
</style>
