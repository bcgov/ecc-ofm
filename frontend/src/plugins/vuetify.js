import '@fortawesome/fontawesome-free/css/all.css';
import '@mdi/font/css/materialdesignicons.css';
import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import { Ripple } from 'vuetify/lib/directives';

Vue.use(Vuetify, {
  directives: {
    Ripple
  }
});

export default new Vuetify({
  theme: {
    light: true,
    dark: false
  },
  icons: {
    iconfont: 'fa',
    values: {
      login: 'fas fa-user-clock',
      fast: 'fas fa-shipping-fast',
      sign_in: 'fas fa-sign-in-alt',
      info1: 'fas fa-info-circle',
      downArrow: 'fas fa-angle-down',
      upArrow: 'fas fa-angle-up',
      user: 'far fa-user',
      copy: 'fas fa-copy',
      search: 'fas fa-search',
      error: 'fas fa-exclamation-triangle',
      lock: 'fas fa-lock',
      info2: 'fas fa-info-circle fa-10x',
      question: 'fas fa-question-circle fa-10x',
      plus: 'fas fa-plus-circle',
      minus: 'fas fa-minus-circle',
    }
  }
});
