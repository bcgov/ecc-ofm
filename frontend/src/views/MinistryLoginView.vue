<template>
  <v-container fluid class="full-height">
    <!-- login article -->
    <!-- <article id="login-banner" class="top-banner"> -->
    <v-row align="center" justify="center">
      <v-col cols="12" align="center" class="pt-0">
        <v-img src="@/assets/images/login-header-img-resized.jpg" max-width="1448" class="align-end">
          <v-col align="center" style="background-color: rgb(28, 115, 158, 0.85)">
            <span class="text-h4 text-white font-weight-bold" style="font-family: BCSans">Early Childhood Care - Operating Funding Model</span>
            <br />
            <span class="text-h6 text-white">Welcome</span>
          </v-col>
        </v-img>
      </v-col>
    </v-row>

    <v-container class="containerWidth1450 pa-0">
      <v-row>
        <v-col class="xs-12 lg-6 xl-2">
          <v-divider></v-divider>
          <v-card flat class="d-flex flex-column justify-center align-center">
            <v-card-title class="gov-header">
              <h4 id="login_text">Ministry</h4>
            </v-card-title>
            <v-card-text id="login_descriptor">Log in with your IDIR account</v-card-text>
            <v-card-actions>
              <v-row>
                <AppButton icon="mdi-login" :href="authRoutes.LOGIN_IDIR" @click="clearStorage" width="250px" id="idir-login">IDIR Log In</AppButton>
              </v-row>
            </v-card-actions>
          </v-card>
        </v-col>
      </v-row>
    </v-container>
    <!-- </article> -->
  </v-container>
</template>

<script>
import { mapActions, mapState } from 'pinia'
import { AuthRoutes } from '@/utils/constants'
import { useAuthStore } from '@/stores/auth'
import AppButton from '@/components/ui/AppButton.vue'

export default {
  name: 'MinistryLoginView',
  components: { AppButton },
  data() {
    return {
      authRoutes: AuthRoutes,
    }
  },
  computed: {
    ...mapState(useAuthStore, ['isAuthenticated']),
  },
  methods: {
    ...mapActions(useAuthStore, ['setJwtToken']),
    clearStorage() {
      this.setJwtToken()
    },
  },
}
</script>

<style scoped>
.login-card {
  max-width: 400px;
}

.gov-header {
  color: #003366;
}

.top-banner {
  min-height: 500px;
  background-size: cover;
  align-items: center;
  display: flex;
}

.full-height {
  height: 100%;
}

.containerWidth1450 {
  max-width: 1450px;
}
</style>
