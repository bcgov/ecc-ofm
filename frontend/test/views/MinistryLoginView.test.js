import { describe, test } from 'vitest'
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

import MinistryLoginView from '@/views/MinistryLoginView.vue'
import { createTestingPinia } from '@pinia/testing'
import { mount } from '@vue/test-utils'

const vuetify = createVuetify({ components, directives })

describe('MinistryLoginView', () => {
  const global = { global: { plugins: [createTestingPinia(), vuetify] } }
  test('init', async () => {
    const wrapper = mount(MinistryLoginView, global)
    expect(wrapper.text()).toContain('IDIR')
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('login', async () => {
    const wrapper = mount(MinistryLoginView, global)

    const loginButton = wrapper.get('#idir-login')
    expect(loginButton.attributes().href).toEqual('/api/auth/login-idir')
  })
})
