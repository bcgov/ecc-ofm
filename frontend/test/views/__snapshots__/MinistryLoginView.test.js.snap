// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MinistryLoginView > init 1`] = `
"<div data-v-18b4d885=\\"\\" class=\\"v-container v-container--fluid v-locale--is-ltr full-height\\">
  <!-- login article -->
  <!-- <article id=\\"login-banner\\" class=\\"top-banner\\"> -->
  <div data-v-18b4d885=\\"\\" class=\\"v-row align-center justify-center\\">
    <div data-v-18b4d885=\\"\\" class=\\"v-col v-col-12 pt-0\\" align=\\"center\\">
      <div data-v-18b4d885=\\"\\" class=\\"v-responsive v-img v-img--booting align-end\\" style=\\"max-width: 1448px;\\">
        <div class=\\"v-responsive__sizer\\"></div>
        <transition-stub name=\\"fade-transition\\" appear=\\"true\\" persisted=\\"false\\" css=\\"true\\"><img class=\\"v-img__img v-img__img--contain\\" src=\\"/src/assets/images/login-header-img-resized.jpg\\" style=\\"display: none;\\"></transition-stub>
        <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
          <!---->
        </transition-stub>
        <!---->
        <!---->
        <!---->
        <div class=\\"v-responsive__content\\">
          <div data-v-18b4d885=\\"\\" class=\\"v-col\\" style=\\"background-color: rgb(28, 115, 158, 0.85);\\" align=\\"center\\"><span data-v-18b4d885=\\"\\" class=\\"text-h4 text-white font-weight-bold\\" style=\\"font-family: BCSans;\\">Early Childhood Care - Operating Funding Model</span><br data-v-18b4d885=\\"\\"><span data-v-18b4d885=\\"\\" class=\\"text-h6 text-white\\">Welcome</span></div>
        </div>
      </div>
    </div>
  </div>
  <div data-v-18b4d885=\\"\\" class=\\"v-container v-locale--is-ltr containerWidth1450 pa-0\\">
    <div data-v-18b4d885=\\"\\" class=\\"v-row\\">
      <div data-v-18b4d885=\\"\\" class=\\"v-col xs-12 lg-6 xl-2\\">
        <hr data-v-18b4d885=\\"\\" class=\\"v-divider v-theme--light\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
        <div data-v-18b4d885=\\"\\" class=\\"v-card v-card--flat v-theme--light v-card--density-default v-card--variant-elevated d-flex flex-column justify-center align-center\\">
          <!---->
          <div class=\\"v-card__loader\\">
            <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
              <!---->
              <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
              <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                <div class=\\"v-progress-linear__indeterminate\\">
                  <div class=\\"v-progress-linear__indeterminate long\\"></div>
                  <div class=\\"v-progress-linear__indeterminate short\\"></div>
                </div>
              </transition-stub>
              <!---->
            </div>
          </div>
          <!---->
          <!---->
          <div data-v-18b4d885=\\"\\" class=\\"v-card-title gov-header\\">
            <h4 data-v-18b4d885=\\"\\" id=\\"login_text\\">Ministry</h4>
          </div>
          <div data-v-18b4d885=\\"\\" class=\\"v-card-text\\" id=\\"login_descriptor\\">Log in with your IDIR account</div>
          <div data-v-18b4d885=\\"\\" class=\\"v-card-actions\\">
            <div data-v-18b4d885=\\"\\" class=\\"v-row\\"><a data-v-d22d1084=\\"\\" data-v-18b4d885=\\"\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text BC-Gov-PrimaryButton\\" style=\\"width: 250px;\\" href=\\"/api/auth/login-idir\\" id=\\"idir-login\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\">IDIR Log In<i data-v-d22d1084=\\"\\" class=\\"mdi-login mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </a></div>
          </div>
          <!---->
          <!----><span class=\\"v-card__underlay\\"></span>
        </div>
      </div>
    </div>
  </div><!-- </article> -->
</div>"
`;
